java-engine/src/test/java/io/getunleash/engine/UnleashEngineTest.java

```diff
@@ -197,45 +239,81 @@ void testMetrics() throws YggdrasilError {
         assertEquals(2, bucket.getToggles().get("Feature.C").getNo());
     }
 
-    @Test
-    void testImpressionData() throws Exception {
-        String features =
-                Files.readString(
-                        Paths.get(
-                                Objects.requireNonNull(
-                                                getClass()
-                                                        .getClassLoader()
-                                                        .getResource("impression-data-tests.json"))
-                                        .toURI()));
-        String featureName = "with.impression.data";
-
+    @ParameterizedTest
+    @CsvSource({
+        "with.impression.data, true",
+        "with.impression.data.false, false",
+        "with.impression.data.undefined, false"
+    })
+    void impressionData_whenFeature_shouldReturn(String featureName, boolean expectedImpressionData)
+            throws Exception {
         assertFalse(engine.shouldEmitImpressionEvent(featureName));
 
-        engine.takeState(features);
+        takeFeaturesFromResource(engine, "impression-data-tests.json");
         Boolean result = engine.isEnabled(featureName, new Context());
         assertNotNull(result);
         assertTrue(result);
-        assertTrue(engine.shouldEmitImpressionEvent(featureName));
+        assertEquals(expectedImpressionData, engine.shouldEmitImpressionEvent(featureName));
     }
 
-    @Test
-    void testImpressionDataFalse() throws Exception {
-        String features =
-                Files.readString(
-                        Paths.get(
-                                Objects.requireNonNull(
-                                                getClass()
-                                                        .getClassLoader()
-                                                        .getResource("impression-data-tests.json"))
-                                        .toURI()));
-        String featureName = "with.impression.data.false";
+    @ParameterizedTest
+    @MethodSource("customStrategiesInput")
+    void customStrategiesRequired_whenNotConfigured_returnsFalse(
+            List<IStrategy> customStrategies,
+            String featureName,
+            Context context,
+            boolean expectedIsEnabled)
+            throws Exception {
+        UnleashEngine customEngine =
+                new UnleashEngine(new YggdrasilFFI("../target/release"), customStrategies);
```

Comment:
Next thing to poke at I think. We should get away from yeeting native paths at the engine soonish

Reply:
Oh, those constructors are not public, they're just for tests. The public ones: https://github.com/Unleash/yggdrasil/blob/5eabc28e6d6c55aea6b293605aae006fa3bda9e2/java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java#L25-L31

Reply:
Okay yeah that's fair!